crates/bevy_ecs/src/schedule_v3/config.rs

```diff
@@ -0,0 +1,618 @@
+        new_system(self).run_if(condition)
+    }
+
+    fn ambiguous_with<M>(self, set: impl IntoSystemSet<M>) -> SystemConfig {
+        new_system(self).ambiguous_with(set)
+    }
+
+    fn ambiguous_with_all(self) -> SystemConfig {
+        new_system(self).ambiguous_with_all()
+    }
+}
+
+impl IntoSystemConfig<()> for SystemConfig {
+    fn into_config(self) -> Self {
+        self
+    }
+
+    fn in_set(mut self, set: impl SystemSet) -> Self {
+        assert!(!set.is_system_type(), "invalid use of system type set");
+        self.graph_info.sets.insert(set.dyn_clone());
+        self
+    }
+
+    fn before<M>(mut self, set: impl IntoSystemSet<M>) -> Self {
+        self.graph_info.dependencies.insert((
+            DependencyEdgeKind::Before,
+            set.into_system_set().dyn_clone(),
+        ));
+        self
+    }
+
+    fn after<M>(mut self, set: impl IntoSystemSet<M>) -> Self {
+        self.graph_info
+            .dependencies
+            .insert((DependencyEdgeKind::After, set.into_system_set().dyn_clone()));
+        self
+    }
+
+    fn run_if<P>(mut self, condition: impl Condition<P>) -> Self {
+        self.conditions.push(new_condition(condition));
+        self
+    }
+
+    fn ambiguous_with<M>(mut self, set: impl IntoSystemSet<M>) -> Self {
+        let set = set.into_system_set();
+        match &mut self.graph_info.ambiguous_with {
+            detection @ Ambiguity::Check => {
+                let mut ambiguous_with = HashSet::new();
+                ambiguous_with.insert(set.dyn_clone());
+                *detection = Ambiguity::IgnoreWithSet(ambiguous_with);
+            }
+            Ambiguity::IgnoreWithSet(ambiguous_with) => {
+                ambiguous_with.insert(set.dyn_clone());
+            }
+            Ambiguity::IgnoreAll => (),
+        }
+
+        self
+    }
+
+    fn ambiguous_with_all(mut self) -> SystemConfig {
+        self.graph_info.ambiguous_with = Ambiguity::IgnoreAll;
+        self
+    }
+}
+
+// only `System<In=(), Out=()>` system objects can be scheduled
+mod sealed {
+    use crate::{
+        schedule_v3::{BoxedSystemSet, SystemSet},
+        system::{BoxedSystem, IntoSystem},
+    };
+
+    use super::{SystemConfig, SystemSetConfig};
+
+    pub trait IntoSystemConfig<Params> {}
+
+    impl<Params, F: IntoSystem<(), (), Params>> IntoSystemConfig<Params> for F {}
+
+    impl IntoSystemConfig<()> for BoxedSystem<(), ()> {}
+
+    impl IntoSystemConfig<()> for SystemConfig {}
+
+    pub trait IntoSystemSetConfig {}
+
+    impl<S: SystemSet> IntoSystemSetConfig for S {}
+
+    impl IntoSystemSetConfig for BoxedSystemSet {}
+
+    impl IntoSystemSetConfig for SystemSetConfig {}
+}
+
+/// A collection of [`SystemConfig`].
+pub struct SystemConfigs {
+    pub(super) systems: Vec<SystemConfig>,
+    pub(super) chained: bool,
+}
+
+/// Types that can convert into a [`SystemConfigs`].
+pub trait IntoSystemConfigs<Params>
```

Comment:
No, because systems are not `Clone`.

Reply:
No, because not everything that implements these traits are `Copy` or `Clone`. 

Also, you probably don't want to dupe the same condition to multiple systems. If those systems are supposed to run as a set, add them to a set, then add the condition to the set itself.